src/xbot/common/math/PID.java

```diff
@@ -23,6 +28,37 @@ public void reset()
     {
         m_prevError = 0;
         m_totalError = 0;
+        errorIsSmall = false;
+        derivativeIsSmall = false;
+    }
+    
+    /**
+     * Set how close the error can be before it is considered
+     * "on-target."
+     *            
+     * This is in the same units as your current and goal values.
+     */
+    public void setErrorTolerance(double errorTolerance) {
```

Comment:
I think we should make our programmatic interfaces as natural as possible, and then take the hit in the dashboard interaction. If we have to special-case a value (whether it's negative numbers, 0, or something else) I'd rather do that in the command or subsystem when taking data from the related property; that way, we can at least minimize bugs when using these APIs in code.